Source/DafnyCore/Backends/GoLang/GoCodeGenerator.cs

```diff
@@ -3024,7 +3033,12 @@ protected class ClassWriter : IClassWriter {
       return (wArray, wRhs);
     }
 
-    protected override string ArrayIndexToInt(string arrayIndex) => $"{HelperModulePrefix}IntOf({arrayIndex})";
+    protected override void EmitArrayIndexToInt(ConcreteSyntaxTree wr, out ConcreteSyntaxTree wIndex) {
```

Comment:
I prefer using a pure function instead of an impure writer one, which would look like:

```c#
protected override ConcreteSyntaxTree ArrayIndexToInt(ConcreteSyntaxTree index) => 
    ConcreteSyntaxTree.Create($@"{HelperModulePrefix}IntOf(""{index}"")");
```

Reply:
I did not know this syntax, but it looks great! I would happily use it. I found how I could take your snippet verbatim and it would work in this case, but looking forward at how this code snippet will be used for the Dafny-to-Dafny compiler, I found it's preferable to leave it as it for now. The Dafny-to-Dafny compiler requires the construction of the outer tree builder before the construction of the inner tree builder and uses fields to keep track of the current tree. I know it's cumbersome and at one point I'd like to revisit that entirely to use more of a pure syntax, but I am not prioritizing that right now.